Helper/Entity/Product/PriceManager/ProductWithChildren.php

```diff
@@ -105,7 +105,7 @@ protected function setFinalGroupPrices($field, $currencyCode, $min, $max, $dashe
         foreach ($this->groups as $group) {
             $groupId = (int) $group->getData('customer_group_id');
 
-            if ($this->customData[$field][$currencyCode]['group_' . $groupId] === 0) {
+            if ((int) $this->customData[$field][$currencyCode]['group_' . $groupId] === 0) {
```

Comment:
Is the `(int)` cast correct? What if the price is `float`, lika `35,76`? Won't it change it to `35` for the future use and therefore it'd index wrong value?

Reply:
Ah! You're right, a float would be more appropriate though to be honest, I'm not sure why we're zero checking here instead of just applying the final group prices.